ui/redux/reducers/memberships.js

```diff
@@ -0,0 +1,197 @@
+};
+
+reducers[ACTIONS.SET_MEMBERSHIP_BUY_STARTED] = (state, action) => {
+  const newPendingBuyIds = new Set(state.pendingBuyIds);
+  newPendingBuyIds.add(action.data);
+  return { ...state, pendingBuyIds: Array.from(newPendingBuyIds) };
+};
+reducers[ACTIONS.SET_MEMBERSHIP_BUY_SUCCESFUL] = (state, action) => {
+  const newPendingBuyIds = new Set(state.pendingBuyIds);
+  newPendingBuyIds.delete(action.data);
+  return { ...state, pendingBuyIds: Array.from(newPendingBuyIds) };
+};
+reducers[ACTIONS.SET_MEMBERSHIP_BUY_FAILED] = (state, action) => {
+  const newPendingBuyIds = new Set(state.pendingBuyIds);
+  newPendingBuyIds.delete(action.data);
+  return { ...state, pendingBuyIds: Array.from(newPendingBuyIds) };
+};
+
+reducers[ACTIONS.SET_MEMBERSHIP_CANCEL_STARTED] = (state, action) => {
+  const newPendingCancelIds = new Set(state.pendingCancelIds);
+  newPendingCancelIds.add(action.data);
+  return { ...state, pendingCancelIds: Array.from(newPendingCancelIds) };
+};
+reducers[ACTIONS.SET_MEMBERSHIP_CANCEL_SUCCESFUL] = (state, action) => {
+  const newPendingCancelIds = new Set(state.pendingCancelIds);
+  newPendingCancelIds.delete(action.data);
+  return { ...state, pendingCancelIds: Array.from(newPendingCancelIds) };
+};
+reducers[ACTIONS.SET_MEMBERSHIP_CANCEL_FAILED] = (state, action) => {
+  const newPendingCancelIds = new Set(state.pendingCancelIds);
+  newPendingCancelIds.delete(action.data);
+  return { ...state, pendingCancelIds: Array.from(newPendingCancelIds) };
+};
+
+reducers[ACTIONS.DELETE_MEMBERSHIP_STARTED] = (state, action) => {
+  const newPendingDeleteIds = new Set(state.pendingDeleteIds);
+  newPendingDeleteIds.add(action.data);
+  return { ...state, pendingDeleteIds: Array.from(newPendingDeleteIds) };
+};
+reducers[ACTIONS.DELETE_MEMBERSHIP_SUCCESFUL] = (state, action) => {
+  const newPendingDeleteIds = new Set(state.pendingDeleteIds);
+  newPendingDeleteIds.delete(action.data);
+  return { ...state, pendingDeleteIds: Array.from(newPendingDeleteIds) };
+};
+reducers[ACTIONS.DELETE_MEMBERSHIP_FAILED] = (state, action) => {
+  const newPendingDeleteIds = new Set(state.pendingDeleteIds);
+  newPendingDeleteIds.delete(action.data);
+  return { ...state, pendingDeleteIds: Array.from(newPendingDeleteIds) };
+};
+
+reducers[ACTIONS.SET_MEMBERSHIP_MINE_DATA] = (state, action) => ({ ...state, membershipMineByKey: action.data });
+
+reducers[ACTIONS.LIST_MEMBERSHIP_DATA] = (state, action) => {
+  const { channelId, list } = action.data;
+  const newMembershipListById = Object.assign({}, state.membershipListById);
+  newMembershipListById[channelId] = list;
+
+  return { ...state, membershipListById: newMembershipListById };
+};
+
+reducers[ACTIONS.MEMBERSHIP_PERK_LIST_COMPLETE] = (state, action) => ({ ...state, membershipPerks: action.data });
+
+reducers[ACTIONS.GET_MEMBERSHIP_TIERS_FOR_CONTENT_STARTED] = (state, action) => {
+  return { ...state };
+};
```

Comment:
:small_red_triangle: If it's not going to do anything for now, just `return state;` to avoid an unnecessary update/render.